src/NUnitFramework/tests/Internal/Filters/NotFilterTests.cs

```diff
@@ -67,7 +67,7 @@ public void ExplicitMatchTest()
         {
             var filter = new NotFilter(new CategoryFilter("Dummy"));
 
-            Assert.False(filter.IsExplicitMatch(_topLevelSuite));
+            Assert.True(filter.IsExplicitMatch(_topLevelSuite));
```

Comment:
I have no problem with that.  It seems to make sense, if that is how Explict is supposed to work.  It is a simple enough change.

 

From: CharliePoole [mailto:notifications@github.com] 
Sent: Saturday, October 03, 2015 9:18 PM
To: nunit/nunit
Cc: Neil Colvin
Subject: Re: [nunit] Fix NotFilter so -exclude works correctly (#862)

 

In src/NUnitFramework/tests/Internal/Filters/NotFilterTests.cs <https://github.com/nunit/nunit/pull/862#discussion_r41093385> :

> @@ -67,7 +67,7 @@ public void ExplicitMatchTest()
>          {
>              var filter = new NotFilter(new CategoryFilter("Dummy"));
>  
> -            Assert.False(filter.IsExplicitMatch(_topLevelSuite));
> +            Assert.True(filter.IsExplicitMatch(_topLevelSuite));

We may need to come back to this in a separate issue since this wasn't really the point of the bug you're fixing.

Explicit was invented to avoid anything being run unless it was specifically selected by the presence of a particular name or category, not by it's absence. So, there's some logic to saying that a not filter should always return false. We would need a bunch of tests to define this clearly though.

—
Reply to this email directly or view it on GitHub <https://github.com/nunit/nunit/pull/862/files#r41093385> .  <https://github.com/notifications/beacon/AFh64acfqMtu3eZkifOcIKfp1SWq2zG1ks5o4HXhgaJpZM4GIcgJ.gif>